sdks/java/core/src/main/java/org/apache/beam/sdk/io/PubsubUnboundedSource.java

```diff
@@ -227,6 +228,9 @@ public long identity() {
    */
   @VisibleForTesting
   static class PubsubCheckpoint<T> implements UnboundedSource.CheckpointMark {
+    /** The {@link SubscriptionPath} to the subscription the reader is reading from. */
+    @Nullable @VisibleForTesting String subscriptionPath;
```

Comment:
Optional comment. Probably more visible like this:

```java
@VisibleForTesting
@Nullable String subscriptionpath;
```

something something clearer for reading.